### PR TITLE
[5.7] Do not save the url in session if the call is prefetch

### PR DESCRIPTION
**Pull request by fkeloks**
## Why ?
I noticed using the new features of `prefetch` that Laravel sessions were working badly.

For example, using this library:  [GoogleChromeLabs/quicklink](https://github.com/GoogleChromeLabs/quicklink), pre-loaded pages modify the session and call the `storeCurrentUrl()` method. Once connected to the site, the `back()` method thus redirects the user to a bad url.

## Solution
To alleviate this problem, I modified the `storeCurrentUrl()` function so that it does not save the current URL if the call is of type `prefetch`.

## Sources
* https://developers.google.com/web/updates/2018/07/nostate-prefetch
* https://medium.com/reloading/preload-prefetch-and-priorities-in-chrome-776165961bbf

### CODE REVIEW

File: src/Illuminate/Http/Request.php [hunk 1]
```diff
@@ -243,6 +243,16 @@ public function pjax()
...
     }
 
+    /**
+     * Determine if the request is the result of an prefetch call.
+     *
+     * @return bool
+     */
+    public function prefetch()
+    {
+        return $this->server->get('HTTP_X_MOZ') === 'prefetch' || $this->headers->get('Purpose') === 'prefetch';
```
Comment (phanan) [inline src/Illuminate/Http/Request.php]: Can this value be case-insensitive?

Comment (fkeloks) [reply src/Illuminate/Http/Request.php]: It seems to me that the header names are not case-sensitive.

Comment (phanan) [reply src/Illuminate/Http/Request.php]: I mean the _value_, not the key.

Comment (fkeloks) [reply src/Illuminate/Http/Request.php]: Oh, excuse me, I did not understand.  
Looking for I have for now never a value that was not in lowercase for this variable.

File: src/Illuminate/Session/Middleware/StartSession.php [hunk 2]
```diff
@@ -41,9 +41,11 @@ public function __construct(SessionManager $manager)
     /**
      * Handle an incoming request.
      *
-     * @param  \Illuminate\Http\Request  $request
-     * @param  \Closure  $next
```
Comment (imanghafoori1) [inline src/Illuminate/Session/Middleware/StartSession.php]: You may add the removed spaces here.

File: src/Illuminate/Session/Middleware/StartSession.php [hunk 3]
```diff
@@ -119,8 +121,10 @@ public function getSession(Request $request)
     /**
      * Remove the garbage from the session if necessary.
      *
-     * @param  \Illuminate\Contracts\Session\Session  $session
+     * @param  \Illuminate\Contracts\Session\Session $session
```
Comment (imanghafoori1) [inline src/Illuminate/Session/Middleware/StartSession.php]: Also here

File: src/Illuminate/Session/Middleware/StartSession.php (hunk 2 above)
Comment (fkeloks) [reply src/Illuminate/Session/Middleware/StartSession.php]: What is the double space for?
I used phpstorm formatting, it is he who deleted this space. If necessary, I will actually re-apply that.

File: src/Illuminate/Http/Request.php (hunk 1 above)
Comment (GrahamCampbell) [reply src/Illuminate/Http/Request.php]: We could use https://github.com/laravel/framework/pull/27001 to compare the strings, if it is accepted.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Pull request closed without merging**